lib/LaTeXML/Core/Document.pm

```diff
@@ -1082,17 +1087,39 @@ sub openText_internal {
   my $qname;
   if ($$self{node}->nodeType == XML_TEXT_NODE) {    # current node already is a text node.
     print STDERR "Appending text \"$text\" to " . Stringify($$self{node}) . "\n" if $LaTeXML::Core::Document::DEBUG;
+    my $parent = $$self{node}->parentNode;
+    if ($LaTeXML::BOX && $parent->getAttribute('_autoopened')) {
+      $self->appendTextBox($parent, $LaTeXML::BOX); }
     $$self{node}->appendData($text); }
   elsif (($text =~ /\S/)                            # If non space
     || $self->canContain($$self{node}, '#PCDATA')) {    # or text allowed here
     my $point = $self->find_insertion_point('#PCDATA');
     my $node  = $$self{document}->createTextNode($text);
+    if ($point->getAttribute('_autoopened')) {
+      $self->appendTextBox($point, $LaTeXML::BOX); }
     print STDERR "Inserting text node for \"$text\" into " . Stringify($point) . "\n"
       if $LaTeXML::Core::Document::DEBUG;
     $point->appendChild($node);
     $self->setNode($node); }
   return $$self{node}; }                                # return the text node (current)
 
+# Since xml text nodes don't have attributes to record the origining box,
+# we need to manage the accumulation of autoOpen'ed boxes
+# Indeed, propogate it to ancestors if they were autoOpened for same cause (box)
+sub appendTextBox {
+  my ($self, $node, $box) = @_;
+  my $origbox = $self->getNodeBox($node);
+  if ($origbox && ($box ne $origbox)) {    # if not already the same box
+    my $newbox = List($origbox, $box);
+    $self->setNodeBox($node, $newbox);
+    my $p = $node;
+    # AND, propogate change to autoOpen'd ancestors based on same initial box
+    while (($p = $p->parentNode) && ($p->nodeType == XML_ELEMENT_NODE)
+      && $p->getAttribute('_autoopened')
+      && (($self->getNodeBox($p) || '') eq $origbox)) {
+      $self->setNodeBox($p, $newbox); } }
+  return; }
+
 # Question: Why do I have math ligatures handled within openMathText_internal,
 # but text ligatures handled within closeText_internal ???
```

Comment:
The above old question about ligatures still sounds curious.

Reply:
but still a good question :>